scripts/pre_commit_linter.py

```diff
@@ -1344,7 +1343,16 @@ def _validate_and_parse_js_files(all_files):
     for filename in files_to_check:
         print 'Validating and parsing %s file ...' % filename
         content = FileCache.read(filename)
-        parsed_js_files[filename] = parser.parse(content)
+
+        content_with_no_non_ascii_characters = ''
```

Comment:
Hm, this seems a bit dodgy -- what's going on here, and why is this necessary?

Reply:
Parsing js files with esprima raises an illegal token exception when non ascii characters like `₹` are present in the file. So, before parsing file contents, all non ascii characters are removed.

Reply:
No, don't do that. Find a way to parse the complete file without removing random characters from it -- maybe you need to encode it properly, or change a setting in esprima, or something. You might consider asking the esprima folks for help if needed, but in general you shouldn't be changing the "meaning" of the file when linting.

Reply:
Sent you a mail outlining instructions to reproduce the error.

Reply:
I was trying to convert characters to their respective unicode form as we discussed but it was giving unexpected end of data error. While searching through, I found a better fix [here](https://stackoverflow.com/a/10487617). The read operation was just returning bytes and decoding the bytes fixed the issue!